### PR TITLE
Add `tensor product representation` properties to UFL element wrappers

### PR DESCRIPTION
**Pull request by IgorBaratta**
Used in `igor/tensor` branch of ffcx.

### CODE REVIEW

File: python/basix/ufl_wrapper.py [hunk 1]
```diff
@@ -351,6 +360,17 @@ def is_custom_element(self) -> bool:
         """True if the element is a custom Basix element."""
         return self._is_custom
 
+    @property
+    def has_tensor_product_factorisation(self) -> bool:
+        """True if the element has a tensor product factorisation."""
+        return self.element.has_tensor_product_factorisation
+
```
Comment (garth-wells) [inline python/basix/ufl_wrapper.py:367]: It’s not clear from the docstring what is returned. Could some more explanation be added?

Comment (garth-wells) [reply python/basix/ufl_wrapper.py:375]: @mscroggs?

Comment (mscroggs) [reply python/basix/ufl_wrapper.py:383]: I've added a bit of detail to the docstring

**Comment by mscroggs** [comment]
https://github.com/FEniCS/dolfinx/pull/2282 should fix the failing DOLFINx integration tests

**Pull request merged**